pandas/core/arrays/integer.py

```diff
@@ -209,13 +207,14 @@ def coerce_to_array(
     elif not (is_integer_dtype(values) or is_float_dtype(values)):
         raise TypeError(f"{values.dtype} cannot be converted to an IntegerDtype")
 
+    if values.ndim != 1:
+        raise TypeError("values must be a 1D list-like")
+
     if mask is None:
-        mask = isna(values)
+        mask = libmissing.is_numeric_na(values)
```

Comment:
Why is it here not needed to check if the mask is consistent with `isna(values)` (as you did for floating) ?

Reply:
bc non-numeric NA values will be caught in safe_cast below